src/main/java/com/epam/aidial/core/service/ResourceService.java

```diff
@@ -0,0 +1,401 @@
+package com.epam.aidial.core.service;
+
+import com.epam.aidial.core.data.MetadataBase;
+import com.epam.aidial.core.data.ResourceFolderMetadata;
+import com.epam.aidial.core.data.ResourceItemMetadata;
+import com.epam.aidial.core.storage.BlobStorage;
+import com.epam.aidial.core.storage.ResourceDescription;
+import com.epam.aidial.core.util.Compression;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import lombok.Getter;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.jclouds.blobstore.domain.Blob;
+import org.jclouds.blobstore.domain.BlobMetadata;
+import org.jclouds.blobstore.domain.PageSet;
+import org.jclouds.blobstore.domain.StorageMetadata;
+import org.jclouds.blobstore.domain.StorageType;
+import org.redisson.api.RMap;
+import org.redisson.api.RScoredSortedSet;
+import org.redisson.api.RedissonClient;
+import org.redisson.client.codec.StringCodec;
+
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import javax.annotation.Nullable;
+
+@Slf4j
+public class ResourceService implements AutoCloseable {
+
+    private static final String BLOB_EXTENSION = ".json";
+    private static final String REDIS_QUEUE = "resource:queue";
+    private static final Set<String> REDIS_FIELDS = Set.of("body", "created_at", "updated_at", "synced", "exists");
+    private static final Set<String> REDIS_FIELDS_NO_BODY = Set.of("created_at", "updated_at", "synced", "exists");
+
+    private final Vertx vertx;
+    private final RedissonClient redis;
+    private final BlobStorage blobStore;
+    private final LockService lockService;
+    @Getter
+    private final int maxSize;
+    private final long syncTimer;
+    private final long syncDelay;
```

Comment:
Can you please add one line comment, why syncDelay is needed.

Reply:
added javadoc